### DISCUSSION TITLE
[Missing Documentation] - Please document in README whether the Amazon Linux 2023 kernel is used for Lambda

### DISCUSSION THREAD

**Discussion by joshtriplett** [amazonlinux/amazon-linux-2022#832] [Q&A]
(Apologies in advance in the likely case that this is the wrong place.)

There are a variety of requests people have for the Lambda and Fargate kernel configuration. Does the `amazon-linux-2023` project provide the kernel for that, or is that owned elsewhere?

Could the README please document whether this is the right place for issues with the Lambda kernel configuration, and if not, please point people towards the correct place for that?

**Comment by stewartsmith** [comment]
(I've moved this over to the discussion area, as that's likely the best place)

The Lambda microVM kernel is a variant of the Amazon Linux one - I do not think the config is published anywhere currently.

Fargate hosts currently run Amazon Linux (AL2 IIRC), I believe with a stock kernel.

**Reply by joshtriplett** [reply]
@stewartsmith Thanks for the response!

Would it be possible to get documentation in the README that Fargate uses a stock Amazon Linux kernel, while Lambda doesn't? (And, ideally, a pointer to somewhere to report bugs or feature requests for the latter?)

**Reply by bryantbiggs** [reply]
Highly unlikely, those are implementation details that customers need not be concerned with and subject to change

**Reply by joshtriplett** [reply]
@bryantbiggs Fair enough. If Amazon Linux is not the place to file *any* such issues, it'd be nice to have a pointer to the right place for that.

**Reply by bryantbiggs** [reply]
could you elaborate a bit on what the motivation is for this information? for AWS Lambda and AWS Fargate, the OS is an implementation detail that is not exposed to users, other than that it is a Linux based OS that supports the listed CPU instruction sets

**Reply by joshtriplett** [reply]
@bryantbiggs There are specific Linux kernel features that'd be incredibly useful for AWS Lambda, such as the ability to mount FUSE filesystems, or have certain `io_uring` capabilities available. Amazon Linux does a great job of having clear roadmaps and issue trackers and the ability to handle feature requests. I was hoping to find the appropriate place to report issues for Lambda in a similar fashion. Since Amazon Linux manages a kernel whose configuration needs to satisfy a wide variety of use cases and customers, and since the Lambda base system seems to heavily leverage Amazon Linux, I hoped that this might be a good starting point to find the appropriate channel. (Apologies for the fact that this itself isn't the right channel; it's difficult to find the right entry point.)

**Reply by bryantbiggs** [reply]
for our container services, we have the publicly available containers roadmap - https://github.com/aws/containers-roadmap/issues

you can see one request related to what you have mentioned here https://github.com/aws/containers-roadmap/issues/412

however, for Lambda, I don't believe we have anything public facing and folks will have to work through their respective AWS account teams to surface those requests and feedback

**Reply by joshtriplett** [reply]
> however, for Lambda, I don't believe we have anything public facing

Alright. That's unfortunate, but I appreciate you looking into it.
